common/slack_notifications.py

```diff
@@ -10,8 +10,40 @@
 VMAAS_ENV = os.getenv('VMAAS_ENV', 'dev')
 
 
-def send_slack_notification(message):
+def format_message(msg: dict):
+    """Msg might be in followed format {header: value, name: cert_name, valid_to: date, expire_in: days}"""
+    formatted = {
+        "blocks": [
+            {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": f"*[{VMAAS_ENV}]* :warning: {msg['header']}! :warning:\n"
+                            f":page_with_curl: *Certificate name*: {msg['name']}"
+                }
+            }
+
+        ]
+    }
+    date = msg.get('date')
+
+    if date:
+        formatted['blocks'].append({"type": "divider"})
+        formatted['blocks'].append({
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f":date: *Valid to*: {date['valid_to']:%Y-%m-%d %H:%M}\n"
+                        f":hourglass_flowing_sand: *Expire in*: {date['expire_in']} days"
+            }
+        })
+
+    return formatted
+
+
+def send_slack_notification(format_callback, msg: dict):
     """If SLACK_WEBHOOK is set, sends notification to Slack"""
+    formatted = format_callback(msg)
```

Comment:
Is there a reason why it's called as callback? Is the callback ever different method than format_message? If not I would call the format_message method directly so we don't have to import the method in repository_controller and pass it as an argument.

Reply:
I was thinking in case we need different formatting in future for another notification